src/main/java/edu/ucdavis/dss/ipa/api/components/budget/BudgetViewController.java

```diff
@@ -157,6 +160,11 @@ public LineItem createLineItem(@PathVariable long budgetScenarioId,
             return null;
         }
 
+        if (lineItemDTO.getTeachingAssignment() != null) {
```

Comment:
What does this do? If the assignment exists, query it from the database and set it to the same value?

Reply:
The deserializer pattern currently used throughout IPA parses the json, and translates a 'teachingAssignmentId' property into a new TeachingAssignment() object and assigns it an id value. Then, if its relevant, the controller can fetch the real entities based on the scaffolds.